### PR TITLE
Identity-operate integration

### PR DESCRIPTION
**Pull request by Zelldon**
Integrates Identity with Operate. 

 * uses the recent release candidate at the time (8.0.0-rc1)
 * Configure Operate and Identity to communicate with each other
 * Make it possible to disable this configuration
 * Auto generate a secret for Operate-Identity ( can be also overwritten by user and an existing secret)
 * Adjust docs, values file
 * Add notes in Helm release notes file
 * Add template tests
 * Add integration test for operate login

New notes:

```
### Operate

As part of the Operate HELM Chart an ingress definition can be deployed, but you require to have an Ingress Controller for that Ingress to be Exposed.
In order to deploy the ingress manifest, set `operate.ingress.enabled` to `true`.
If you don't have an Ingress Controller you can use kubectl port-forward to access Operate from outside the cluster:

> kubectl port-forward svc/camunda-platform-test-operate 8080:80

Per default authentication via Identity/Keycloak is enabled. In order to login into Operate please port-forward to Keycloak
as well, otherwise a login will not be possible. Make sure you use `18080` as port.

> kubectl port-forward svc/camunda-platform-tes 18080:80

Now you can point your browser to `http://localhost:8080`

Default user and password: "demo/demo"

```

@dlavrenuek maybe you can shortly check whether the config is like we discuss ( I would expect, just to be sure)
@npepinpe please do a normal review :)

### CODE REVIEW

File: charts/camunda-platform/templates/NOTES.txt
Comment (Zelldon) [inline charts/camunda-platform/templates/NOTES.txt:92]: Keycloak has the weird property to cap the names at 20 chars, the reason is that wildfly only supports a certain amount of characters :shrug: :laughing:

**Review by npepinpe — APPROVED** [review]
:rocket: 

@dlavrenuek - I didn't really check if it's correctly configured or working properly, just that it starts. I think you will be better placed to check that it works as expected.

Other notes:

- :question: The version is 8.0.0-rc1, I guess we'll bump this to 8.0.0 soon? Everything should be available now.
- :question: There are 4 configuration settings for the integration, but only two of them are tested in the deployment test, and one in the secret test - any reason for not testing the last one about the root URL? Or maybe I missed it.

Anyway, no blockers that I could see

File: charts/camunda-platform/README.md
Comment (Zelldon) [reply charts/camunda-platform/README.md:66]: Not sure what you mean.

Maybe to explain it better:

The user can have a secret manifest deploy on there namespace which should contain the `operate-secret ` if it is referenced here. Does this make more sense? What should/could I add here to be more specific?

File: charts/camunda-platform/charts/operate/templates/deployment.yaml
Comment (Zelldon) [reply charts/camunda-platform/charts/operate/templates/deployment.yaml:29]: Currently not. https://github.com/camunda/camunda-platform-helm/issues/256

File: charts/camunda-platform/test/identity/deployment_test.go
Comment (Zelldon) [reply charts/camunda-platform/test/identity/deployment_test.go:359]: Causes in the controller also interesting code

**Comment by Zelldon** [comment]
Thanks for your review @npepinpe :+1: 

Regarding your questions:

> Other notes:
> 
> :question:  The version is 8.0.0-rc1, I guess we'll bump this to 8.0.0 soon? Everything should be available now.

I will do it after I merged all PR's, otherwise we have to many conflicts.

> :question:  There are 4 configuration settings for the integration, but only two of them are tested in the deployment test, and one in the secret test - any reason for not testing the last one about the root URL? Or maybe I missed it.
> 

The others are tested via golden files. In the template tests (in deployment test) we only test conditional properties.

I will merge it after @dlavrenuek gives us the go :)

File: charts/camunda-platform/templates/NOTES.txt
Comment (Zelldon) [reply charts/camunda-platform/templates/NOTES.txt:57]: Ok sounds legit I will adjust it thanks :)

File: charts/camunda-platform/README.md
Comment (Zelldon) [reply charts/camunda-platform/README.md]: I updated the doc

Comment (npepinpe) [reply charts/camunda-platform/README.md]: I guess it's not clear to me what the secret format should be. It's clearer with the example though. So, I need to create a secret which will has a field with key `operate-secret` and the value should be...?

Comment (Zelldon) [reply charts/camunda-platform/README.md]: The value is the secret :sweat_smile: it doesn;t matter it is just a string which will be used for communication/authentication between identity and operate.

Comment (Zelldon) [reply charts/camunda-platform/README.md]: https://c8.docs.camunda.io/docs/self-managed/identity/deployment/configuration-variables/#component-configuration

File: charts/camunda-platform/charts/identity/templates/deployment.yaml
Comment (Zelldon) [reply charts/camunda-platform/charts/identity/templates/deployment.yaml:60]: Yeah TBH I thought it is not necessary as normal User to access identity :shrug: It worked for me with just port-forward to keycloak and operate. Is it not expected to work like that? :D

Comment (Zelldon) [reply charts/camunda-platform/charts/identity/templates/deployment.yaml:60]: BTW shouldn't be a problem as far as I can see, since the user need to port-forward any way the traffic and he have to choose a free local port right?

File: charts/camunda-platform/README.md
Comment (Zelldon) [reply charts/camunda-platform/README.md:67]: Actually it is optional, which is why I put as `can be`. Even if you use an ingress you don't need to configure it :sweat_smile:

**Comment by Zelldon** [comment]
Thanks for your reviews I will merge this. I think we can continuously improve if we see anything.

**Pull request merged**
